inc/c_util/two_d_array_ll.h

```diff
@@ -15,10 +15,236 @@
+    /* Codes_SRS_TWO_D_ARRAY_07_001: [ If row_size equals to zero, TWO_D_ARRAY_CREATE(T) shall fail and return NULL. ]*/                                                                           \
+    if (row_size == 0 ||                                                                                                                                                                           \
+    /* Codes_SRS_TWO_D_ARRAY_07_002: [ If col_size equals to zero, TWO_D_ARRAY_CREATE(T) shall fail and return NULL. ]*/                                                                           \
+       col_size == 0)                                                                                                                                                                              \
+    {                                                                                                                                                                                              \
+        LogError("Invalid arguments: uint64_t row_size=%" PRIu64 ", col_size = %" PRIu64, row_size, col_size);                                                                                    \
+    }                                                                                                                                                                                              \
+    else                                                                                                                                                                                           \
+    {                                                                                                                                                                                              \
+        /* Codes_SRS_TWO_D_ARRAY_07_003: [ TWO_D_ARRAY_CREATE(T) shall call THANDLE_MALLOC_FLEX with TWO_D_ARRAY_FREE(T) as dispose function, nmemb set to row_size and size set to sizeof(T*). ]*/\
+        result = THANDLE_MALLOC_FLEX(TWO_D_ARRAY_TYPEDEF_NAME(C))(TWO_D_ARRAY_LL_FREE_NAME(C), row_size, sizeof(T*));                                                                              \
+        /* Codes_SRS_TWO_D_ARRAY_07_005: [ If there are any errors then TWO_D_ARRAY_CREATE(T) shall fail and return NULL. ]*/                                                                      \
+        if (result == NULL)                                                                                                                                                                        \
+        {                                                                                                                                                                                          \
+            LogError("failure in malloc_flex(sizeof(TWO_D_ARRAY_HANDLE)=%zu, row_size=%" PRIu64 ", sizeof(unsigned char*)=%zu);", sizeof(T*), row_size, sizeof(unsigned char*));                   \
```

Comment:
> sizeof(unsigned char*) [](http://example.com/codeflow?start=98&length=23)

This should be sizeof(T*)

Reply:
And we should print the T with MU_TO_STRING) so that the log line has information about what we're working with.

Reply:
done